app/grub_control.py

```diff
@@ -472,18 +453,15 @@ def set_next_boot_entry(self, menuentry_no):
             return False
         return True
 
-    def set_next_bank_boot(self, no_submenu=True):
+    def set_next_bank_boot(self):
         """
         set next boot grub menue entry to custom config menu
         """
         # get grub.cfg menuentries
-        if no_submenu:
-            menu_entries = self.count_grub_menue_entries_wo_submenu(self._grub_cfg_file)
-        else:
-            menu_entries = self.count_grub_menue_entries(self._grub_cfg_file)
-        if menu_entries > 0:
+        menuentries = self.count_grub_menu_entries(self._grub_cfg_file)
+        if menuentries > 0:
             # set next boot menuentry to custum menuentry
-            res = self.set_next_boot_entry(menu_entries)
+            res = self.set_next_boot_entry(menuentries)
```

Comment:
下記ぐらいでいいと思っています。そうするとごっそり関連する処理が削れると思います。
```suggestion
            menus = GrubCfgParser(open(self._grub_cfg_file).read()).parse()
            res = self.set_next_boot_entry(len(menus))
```

Reply:
変更しました